sklearn/preprocessing/data.py

```diff
@@ -641,7 +643,14 @@ def partial_fit(self, X, y=None):
                 # First pass
                 if not hasattr(self, 'n_samples_seen_'):
                     self.mean_, self.var_ = mean_variance_axis(X, axis=0)
-                    self.n_samples_seen_ = X.shape[0]
+                    self.n_samples_seen_ = (np.ones(X.shape[1], dtype=np.int32)
+                                            * X.shape[0])
+                    print(self.n_samples_seen_)
+                    counts_nan = sparse.csr_matrix(
+                        (np.isnan(X.data), X.indices, X.indptr)).sum(
```

Comment:
this needs a shape kwarg to correctly infer the number of columns. That's the source of the errors for me.